questions/126/explanation.md

```diff
@@ -1,3 +1,27 @@
-§[basic.lookup.unqual]¶13 states "A name used in the definition of a static data member of class `X` (...) is looked up as if the name was used in a member function of `X`."
+Will the global `foo` or the member `foo` be chosen?
 
-Even though the call `foo()` occurs outside the class, since `foo` is used in the definition of the static data member `foobar::x`, it is looked up as if `foo()` was called in a member function of `foobar`. If `foo()` was called in a member function of `foobar`, `foobar::foo()` would be called, not the global `foo()`.
\ No newline at end of file
+First, notice that the name `foo` is unqualified. §[basic.lookup.unqual]¶4:
+
+> An *unqualified name* is a name that does not immediately follow a *nested-name-specifier* or the `.` or `->` in a class member access expression (§[expr.ref]), possibly after a `template` keyword or `~`. Unless otherwise specified, such a name undergoes unqualified name lookup from the point where it appears.
+
+What does unqualified name lookup mean? §[basic.lookup.unqual]¶3:
+
+> *Unqualified name lookup* from a program point performs an unqualified search in its immediate scope.
+
+The definition of an unqualified search is given in §[basic.lookup.unqual]¶2:
+
+> An *unqualified search* in a scope `S` from a program point `P` includes the results of searches from `P` in
+>
+> — `S`, and
+>
+> — for any scope `U` that contains `P` and is or is contained by `S`, each namespace contained by `S` that is nominated by a *using-directive* that is active in `U` at `P`.
+>
+> If no declarations are found, the results of the unqualified search are the results of an unqualified search in the parent scope of `S`, if any, from `P`.
+
+So, since we don't have any using directives, we search for `foo` in its immediate scope and if we find any result, we stop searching.
```

Comment:
```suggestion
So, since we don't have any using directives, we search for `foo` in its immediate scope and if we find any result, we stop searching. (If we don't, we keep searching in parent scopes, recursively.)
```

I think it might help to add this, to clarify that the immediate scope is not the only scope we search.